sdk/eventhub/azure-eventhub/azure/eventhub/_buffered_producer/_buffered_producer.py

```diff
@@ -0,0 +1,221 @@
+                    batch = self._buffered_queue.get()
+                    if not batch:
+                        self._max_concurrent_sends_semaphore.release()
+                        continue
+                    try:
+                        _LOGGER.info("Partition {} is sending.".format(self.partition_id))
+                        self._producer.send(
+                            batch,
+                            timeout=timeout_time - time.time() if timeout_time else None
+                        )
+                        self._on_success(batch._internal_events, self.partition_id)
+                        _LOGGER.info(
+                            "Partition {} sending {} events succeeded.".format(
+                                self.partition_id, len(batch)
+                            )
+                        )
+                    except Exception as exc:
+                        self._on_error(batch._internal_events, exc, self.partition_id)
+                        _LOGGER.info(
+                            "Partition {} sending {} events failed due to exception: {!r} ".format(
+                                self.partition_id, len(batch), exc
+                            )
+                        )
+                    finally:
+                        self._cur_buffered_len -= len(batch)
+                        self._max_concurrent_sends_semaphore.release()
+                        self._not_full.notify()
+                # If flush could not get the semaphore, we log and raise error if wanted
+                else:
+                    _LOGGER.info(
+                        "Partition {} fails to flush due to timeout.".format(
+                            self.partition_id
+                        )
+                    )
+                    if raise_error:
+                        raise OperationTimeoutError("Failed to flush {!r}".format(self.partition_id))
+        # after finishing flushing, reset cur batch and put it into the buffer
+        self._last_send_time = time.time()
+        self._cur_batch = EventDataBatch(self._max_message_size_on_link)
+        self._buffered_queue.put(self._cur_batch)
+        _LOGGER.info("Partition {} finished flushing.".format(self.partition_id))
+
+    def check_max_wait_time_worker(self):
+        while self._running:
+            with self._not_empty:
+                if not self._buffered_queue.qsize():
+                    _LOGGER.info("Partition {} worker is awaiting data.".format(self.partition_id))
+                    self._not_empty.wait()
+                now_time = time.time()
+                _LOGGER.info("Partition {} worker is checking max_wait_time.".format(self.partition_id))
+                if now_time - self._last_send_time > self._max_wait_time:
+                    # in the worker, not raising error for flush, users can not handle this
+                    self.flush(raise_error=False)
+                    self._last_send_time = now_time
```

Comment:
I think this is already being set in `flush`

Reply:
you're right! removed